{
  "configurations": {
    "CVE_data_version": "4.0",
    "nodes": [
      {
        "cpe_match": [
          {
            "cpe23Uri": "cpe:2.3:a:python:python:2.7:*:*:*:*:*:*:*",
            "vulnerable": true
          },
          {
            "cpe23Uri": "cpe:2.3:a:python:python:3.5:*:*:*:*:*:*:*",
            "vulnerable": true
          },
          {
            "cpe23Uri": "cpe:2.3:a:python:python:3.6:*:*:*:*:*:*:*",
            "vulnerable": true
          },
          {
            "cpe23Uri": "cpe:2.3:a:python:python:3.7:*:*:*:*:*:*:*",
            "vulnerable": true
          },
          {
            "cpe23Uri": "cpe:2.3:a:python:python:*:*:*:*:*:*:*:*",
            "versionEndIncluding": "3.8.0b1",
            "versionStartIncluding": "3.8.0a4",
            "vulnerable": true
          }
        ],
        "operator": "OR"
      },
      {
        "cpe_match": [
          {
            "cpe23Uri": "cpe:2.3:o:redhat:enterprise_linux:7.0:*:*:*:*:*:*:*",
            "vulnerable": true
          }
        ],
        "operator": "OR"
      }
    ]
  },
  "cve": {
    "CVE_data_meta": {
      "ASSIGNER": "cve@mitre.org",
      "ID": "CVE-2019-10160"
    },
    "affects": {
      "vendor": {
        "vendor_data": [
          {
            "product": {
              "product_data": [
                {
                  "product_name": "python",
                  "version": {
                    "version_data": [
                      {
                        "version_affected": "=",
                        "version_value": "2.7"
                      },
                      {
                        "version_affected": "=",
                        "version_value": "3.5"
                      },
                      {
                        "version_affected": "=",
                        "version_value": "3.6"
                      },
                      {
                        "version_affected": "=",
                        "version_value": "3.7"
                      },
                      {
                        "version_affected": "=",
                        "version_value": "3.8.0b1"
                      }
                    ]
                  }
                }
              ]
            },
            "vendor_name": "python"
          },
          {
            "product": {
              "product_data": [
                {
                  "product_name": "enterprise_linux",
                  "version": {
                    "version_data": [
                      {
                        "version_affected": "=",
                        "version_value": "7.0"
                      }
                    ]
                  }
                }
              ]
            },
            "vendor_name": "redhat"
          }
        ]
      }
    },
    "data_format": "MITRE",
    "data_type": "CVE",
    "data_version": "4.0",
    "description": {
      "description_data": [
        {
          "lang": "en",
          "value": "A security regression of CVE-2019-9636 was discovered in python since commit d537ab0ff9767ef024f26246899728f0116b1ec3 affecting versions 2.7, 3.5, 3.6, 3.7 and from v3.8.0a4 through v3.8.0b1, which still allows an attacker to exploit CVE-2019-9636 by abusing the user and password parts of a URL. When an application parses user-supplied URLs to store cookies, authentication credentials, or other kind of information, it is possible for an attacker to provide specially crafted URLs to make the application locate host-related information (e.g. cookies, authentication data) and send them to a different host than where it should, unlike if the URLs had been correctly parsed. The result of an attack may vary based on the application."
        }
      ]
    },
    "problemtype": {
      "problemtype_data": [
        {
          "description": [
            {
              "lang": "en",
              "value": "CWE-255"
            }
          ]
        }
      ]
    },
    "references": {
      "reference_data": [
        {
          "name": "openSUSE-SU-2019:1906",
          "refsource": "SUSE",
          "tags": [],
          "url": "http://lists.opensuse.org/opensuse-security-announce/2019-08/msg00042.html"
        },
        {
          "name": "RHSA-2019:1587",
          "refsource": "REDHAT",
          "tags": [],
          "url": "https://access.redhat.com/errata/RHSA-2019:1587"
        },
        {
          "name": "RHSA-2019:1700",
          "refsource": "REDHAT",
          "tags": [],
          "url": "https://access.redhat.com/errata/RHSA-2019:1700"
        },
        {
          "name": "RHSA-2019:2437",
          "refsource": "REDHAT",
          "tags": [],
          "url": "https://access.redhat.com/errata/RHSA-2019:2437"
        },
        {
          "name": "https://bugzilla.redhat.com/show_bug.cgi?id=CVE-2019-10160",
          "refsource": "CONFIRM",
          "tags": [
            "Issue Tracking",
            "Third Party Advisory"
          ],
          "url": "https://bugzilla.redhat.com/show_bug.cgi?id=CVE-2019-10160"
        },
        {
          "name": "https://github.com/python/cpython/commit/250b62acc59921d399f0db47db3b462cd6037e09",
          "refsource": "CONFIRM",
          "tags": [
            "Patch",
            "Third Party Advisory"
          ],
          "url": "https://github.com/python/cpython/commit/250b62acc59921d399f0db47db3b462cd6037e09"
        },
        {
          "name": "https://github.com/python/cpython/commit/8d0ef0b5edeae52960c7ed05ae8a12388324f87e",
          "refsource": "CONFIRM",
          "tags": [
            "Patch",
            "Third Party Advisory"
          ],
          "url": "https://github.com/python/cpython/commit/8d0ef0b5edeae52960c7ed05ae8a12388324f87e"
        },
        {
          "name": "https://github.com/python/cpython/commit/f61599b050c621386a3fc6bc480359e2d3bb93de",
          "refsource": "CONFIRM",
          "tags": [
            "Patch",
            "Third Party Advisory"
          ],
          "url": "https://github.com/python/cpython/commit/f61599b050c621386a3fc6bc480359e2d3bb93de"
        },
        {
          "name": "https://github.com/python/cpython/commit/fd1771dbdd28709716bd531580c40ae5ed814468",
          "refsource": "CONFIRM",
          "tags": [
            "Patch",
            "Third Party Advisory"
          ],
          "url": "https://github.com/python/cpython/commit/fd1771dbdd28709716bd531580c40ae5ed814468"
        },
        {
          "name": "[debian-lts-announce] 20190625 [SECURITY] [DLA 1834-1] python2.7 security update",
          "refsource": "MLIST",
          "tags": [],
          "url": "https://lists.debian.org/debian-lts-announce/2019/06/msg00022.html"
        },
        {
          "name": "FEDORA-2019-7df59302e0",
          "refsource": "FEDORA",
          "tags": [],
          "url": "https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/2ORNTF62QPLMJXIQ7KTZQ2776LMIXEKL/"
        },
        {
          "name": "FEDORA-2019-7723d4774a",
          "refsource": "FEDORA",
          "tags": [],
          "url": "https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/44TS66GJMO5H3RLMVZEBGEFTB6O2LJJU/"
        },
        {
          "name": "FEDORA-2019-60a1defcd1",
          "refsource": "FEDORA",
          "tags": [],
          "url": "https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/HQEQLXLOCR3SNM3AA5RRYJFQ5AZBYJ4L/"
        },
        {
          "name": "FEDORA-2019-9bfb4a3e4b",
          "refsource": "FEDORA",
          "tags": [],
          "url": "https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/KRYFIMISZ47NTAU3XWZUOFB7CYL62KES/"
        },
        {
          "name": "https://python-security.readthedocs.io/vuln/urlsplit-nfkc-normalization2.html",
          "refsource": "MISC",
          "tags": [
            "Third Party Advisory"
          ],
          "url": "https://python-security.readthedocs.io/vuln/urlsplit-nfkc-normalization2.html"
        },
        {
          "name": "https://security.netapp.com/advisory/ntap-20190617-0003/",
          "refsource": "CONFIRM",
          "tags": [],
          "url": "https://security.netapp.com/advisory/ntap-20190617-0003/"
        }
      ]
    }
  },
  "impact": {
    "baseMetricV2": {
      "acInsufInfo": false,
      "cvssV2": {
        "accessComplexity": "LOW",
        "accessVector": "NETWORK",
        "authentication": "NONE",
        "availabilityImpact": "NONE",
        "baseScore": 5,
        "confidentialityImpact": "PARTIAL",
        "integrityImpact": "NONE",
        "vectorString": "AV:N/AC:L/Au:N/C:P/I:N/A:N",
        "version": "2.0"
      },
      "exploitabilityScore": 10,
      "impactScore": 2.9,
      "obtainAllPrivilege": false,
      "obtainOtherPrivilege": false,
      "obtainUserPrivilege": false,
      "severity": "MEDIUM",
      "userInteractionRequired": false
    },
    "baseMetricV3": {
      "cvssV3": {
        "attackComplexity": "LOW",
        "attackVector": "NETWORK",
        "availabilityImpact": "HIGH",
        "baseScore": 9.8,
        "baseSeverity": "CRITICAL",
        "confidentialityImpact": "HIGH",
        "integrityImpact": "HIGH",
        "privilegesRequired": "NONE",
        "scope": "UNCHANGED",
        "userInteraction": "NONE",
        "vectorString": "CVSS:3.0/AV:N/AC:L/PR:N/UI:N/S:U/C:H/I:H/A:H",
        "version": "3.0"
      },
      "exploitabilityScore": 3.9,
      "impactScore": 5.9
    }
  },
  "lastModifiedDate": "2019-06-17T19:15Z",
  "publishedDate": "2019-06-07T18:29Z"
}